### PR TITLE
Docusuarus versioning (with Pester v4 and v5 cut)

### PR DESCRIPTION
**Pull request by bravo-kernel**
This change was awaiting docusaurus 2.0.0-alpha.62 which brings improved versioning, suited for Pester v4 and v5.

## Preview links

- [v5 docs](https://deploy-preview-54--pester-docs.netlify.app/docs/quick-start) - current version
- [v4 docs](https://deploy-preview-54--pester-docs.netlify.app/docs/V4/quick-start) - frozen version

## Changes

- browsing to `docs` no longer throws a 404 but redirects to `docs/quick-start` (v5 only for now, awaits a fix)

## Versioning Logic

- v4 documentation now lives under https://pester.dev/docs/v4/quick-start
- v4 documentation is copied/frozen into `versioned_docs/version-v4 ` folder
- v4 updates should be done in that folder
- v5 is now the actual/current/live version living inside the expected `docs` folder
- v5 documentation updates should be done there

Users browsing v4 documentation will see the following warning (message autogenerated by D2 using the `v4` version name):

image

## TODO

- [x] update links in the v4 folder to point to the v4 website
- [x] generate new Command Reference using Pester v5 Get-Help (I will do that)
- [x] enable version dropdown
- [ ] move Pester README into website pages
- [ ] update v5 content/pages, upgrade guide, etc.

## Good to Know

- google indexing will not break until we start replacing, renaming v5 pages
- thus, indexing will break for (certain?) Get-Help command reference pages once we update those

## Nice to Have

- It might be worth considering adding a version dropdown for users, a-la this screenshot from the D2 website

    image

### CODE REVIEW

**Comment by nohwnd** [comment]
Did you read my mind? Or did I push the changes for v5.0 docs this weekend? I was just working on adding v5 docs section.

**Comment by bravo-kernel** [comment]
This was awaiting for some time. I have sent you a Discord DM as well. Perhaps discuss there.

File: docs/commands/AfterAll.mdx [hunk 1]
```diff
@@ -9,7 +9,6 @@ keywords:
   - Documentation
 hide_title: false
 hide_table_of_contents: false
-custom_edit_url: null
```
Comment (nohwnd) [inline docs/commands/AfterAll.mdx:12]: this should probably remain in place?

File: docs/commands/AfterAll.mdx [hunk 2]
```diff
@@ -69,8 +68,4 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## RELATED LINKS
 
-[https://pester.dev/docs/commands/BeforeAll](https://pester.dev/docs/commands/BeforeAll)
-
-## EDIT THIS PAGE
```
[the remark this answers is not in the document]
Comment (bravo-kernel) [reply docs/commands/AfterAll.mdx:74]: Good catch, just re-added the `-AppendMarkdown` parameter back to the generate-command-reference script.

File: docs/commands/AfterAll.mdx (hunk 1 above)
Comment (bravo-kernel) [reply docs/commands/AfterAll.mdx]: Good catch, just re-added the `-EditUrl` parameter back to the generate-command-reference script.

**Comment by bravo-kernel** [comment]
These Pester v5 source files should be checked to see what is different in their Get-Help definitions compared to v4.

Links removed from the markdown for now so the build no longer breaks and we can merge.

image

**Comment by bravo-kernel** [comment]
@nohwnd this should be good to merge now

**Comment by nohwnd** [comment]
Will do, when I have some more time to work on this, in case anything would break after the merge, and also so I can clean up a bit :)

**Comment by bravo-kernel** [comment]
👍

**Comment by nohwnd** [comment]
Genius! Thanks!

**Pull request merged**